tests/test_dev_server.py

```diff
@@ -49,10 +48,87 @@ def test_my_lambda_dev_handler():
     assert path == "/t/{id}"
     assert params == {"id": "123"}
 
+    path, params = handler._get_route_params("/?=1&xx=2")  # pylint: disable=protected-access
+    assert path == "/"
+    assert params is None
+
+    with pytest.raises(ValueError):
+        handler._get_route_params("")  # pylint: disable=protected-access
+
+
+@pytest.mark.parametrize(
+    "method", ["do_GET", "do_PATCH", "do_POST", "do_PUT", "do_DELETE", "do_OPTIONS"]
+)
+def test_handle_request(method) -> None:
+    with patch("socket.socket") as msocket:
+        msocket.makefile = lambda a, b: io.BytesIO(b"GET / HTTP/1.1\r\n")
+        msocket.rfile.close = lambda: 0
+        handler = MockedHandler(msocket, ("127.0.0.1", 8888), BaseServer)
+        with patch.object(handler, "handle_request") as mock_handler:
+            getattr(handler, method)()
+            mock_handler.assert_called_once()
+
+
+def test_handle_favicon() -> None:
+    with patch("socket.socket") as msocket:
+        msocket.makefile = lambda a, b: io.BytesIO(b"GET /favicon.ico HTTP/1.1\r\n")
+        msocket.rfile.close = lambda: 0
+        handler = MyLambdaDevHandlerHelloWorld(msocket, ("127.0.0.1", 8888), BaseServer)
+    assert handler.handle_request() is None
+
+
+def test_handle_txt() -> None:
+    with patch.object(MyLambdaDevHandler, "_send_json") as mocked_send:
+        with patch("socket.socket") as msocket:
+            msocket.makefile = lambda a, b: io.BytesIO(b"GET /txt HTTP/1.1\r\n")
+            msocket.rfile.close = lambda: 0
+            MyLambdaDevHandlerHelloWorld(msocket, ("127.0.0.1", 8888), BaseServer)
+
+        mocked_send.assert_has_calls(
+            [
+                call(200, "HelloWorld", {"Content-Type": "text/plain"}),
+                call(
+                    500,
+                    {"error": "Server error"},
+                    headers={"Content-Type": "application/json;charset=UTF-8"},
+                ),
+            ]
+        )
+
 
-def test_my_dev_server():
-    dev_serv = MyDevServer(HelloWorld)
+def test_handle_json() -> None:
+    byte_string = (
+        b'POST /p HTTP/1.1\r\nContent-Length: 8\r\nContent-Type: application/json\r\n\r\n{"a": 1}'
+    )
+    with patch.object(MyLambdaDevHandler, "_send_json") as mocked_send:
+        with patch("socket.socket") as msocket:
+            msocket.makefile = lambda a, b: io.BytesIO(byte_string)
+            msocket.rfile.close = lambda: 0
+            MyLambdaDevHandlerHelloWorld(msocket, ("127.0.0.1", 8888), BaseServer)
+
+        mocked_send.assert_has_calls(
```

Comment:
In both `test_handle_txt` and `test_handle_json`, you are checking if one of the mocked/protected methods was triggered according to the needs. In general, that is not what the test should test 😐  You should rather compare the results which you are getting.